test/passing/break_cases.ml

```diff
@@ -0,0 +1,35 @@
+let f x = function
+  | C
+   |P (this, test, [is; wide; enough; _to; break], [the; line])
+   |A | K -> 1
+  | D ->
+      let a = "this" in
+      let b = "breaks" in
+      ()
+
+let f =
+  let g = function
+    | H when x y <> k -> 2
+    | T | P | U -> 3
+  in
+  fun x g t h y u ->
+    match x with
+    | E -> 4
+    | Z | P | M ->
+      match y with
+      | O -> 5
+      | P when h x ->
+          function
+          | A -> 6
+
+;; match x with
+   | true -> (
+     match y with
+     | true -> "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+     | false -> "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" )
+   | false -> "cccccccccccccccccccccccccccccc"
+
+;; match x with
+   | "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", yyyyyyyyyy
+     when fffffffffffffff bbbbbbbbbb yyyyyyyyyy -> ()
```

Comment:
Without `--break-cases`, if the pattern + when guard if present does not fit on a line, then a break is added after `->` in order to avoid "hiding" a short body at the end of a multi-line pattern. Similarly, hit by reformatting ocamlformat but not shown by the tests, if the pattern doesn't fit, then the when guard is bumped to the next line. Can you confirm that the present behavior with less breaking is as you intend?

Reply:
Yes I intended to limit the breaks. When this option is set, the break is forced for `|` so as you state the guard is bumped to the next line, no matter the size of the pattern.
So we have:
```
| K -> x
| A
  when b -> y
```

I prefer having this : 
```
| K -> x
| A when b -> y
```

But if you insist of having the first formatting I just have to revert my last commit.

Reply:
I'm not insisting, just checking that the behavior is as intended. But to be more explicit, depending on the margin, we will see e.g.:
```
;; match x with
   | ( "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
     , yyyyyyyyyy )
     when fffffffffffffff bbbbbbbbbb yyyyyyyyyy -> ()
```
and
```
;; match x with
   | ( "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
     , yyyyyyyyyy ) when f y -> ()
```
In these sorts of cases, there are possibly less breaks with `--break-cases` than without, which is what makes me wonder if this is intended as it does not fit with my understanding of the informal intent of this option.

Reply:
I have seen cases where people had a harder time reading code because of such hidden when clauses or short bodies, so I would suggest keeping those breaks that the current implementation inserts. But to achieve the output you want otherwise might require adding more boxes, if changing the `fmt` you mention doesn't do it.